python/mujinwebstackclient/webstackclientutil.py

```diff
@@ -0,0 +1,295 @@
+        return super(QueryResult, self).extend(items)
+
+    def insert(self, index, item):
+        self._ensureCompleteQueryResult()
+        return super(QueryResult, self).insert(index, item)
+
+    def index(self, item, start=0, end=None):
+        self._ensureCompleteQueryResult()
+        if end is None:
+            end = len(self)
+        return super(QueryResult, self).index(item, start, end)
+
+    def pop(self):
+        self._ensureCompleteQueryResult()
+        return super(QueryResult, self).pop()
+
+    def count(self, item):
+        self._ensureCompleteQueryResult()
+        return super(QueryResult, self).count(item)
+
+    def remove(self, item):
+        self._ensureCompleteQueryResult()
+        return super(QueryResult, self).remove(item)
+
+    def reverse(self):
+        self._ensureCompleteQueryResult()
+        return super(QueryResult, self).reverse()
+
+    def sort(self, reverse=False, key=None):
+        self._ensureCompleteQueryResult()
+        return super(QueryResult, self).sort(reverse=reverse, key=key)
+
+    def __iadd__(self, items):
+        self._ensureCompleteQueryResult()
+        return super(QueryResult, self).__iadd__(items)
+
+    def __add__(self, items):
+        self._ensureCompleteQueryResult()
+        return super(QueryResult, self).__add__(items)
+
+    def __rmul__(self, value):
+        self._ensureCompleteQueryResult()
+        return super(QueryResult, self).__rmul__(value)
+
+    def __mul__(self, value):
+        self._ensureCompleteQueryResult()
+        return super(QueryResult, self).__mul__(value)
+
+    def __imul__(self, value):
+        self._ensureCompleteQueryResult()
+        return super(QueryResult, self).__imul__(value)
+
+    def __reversed__(self):
+        self._ensureCompleteQueryResult()
+        return super(QueryResult, self).__reversed__()
+
+    def __contains__(self, item):
+        self._ensureCompleteQueryResult()
+        return super(QueryResult, self).__contains__(item)
+
+    def __delitem__(self, index):
+        self._ensureCompleteQueryResult()
+        return super(QueryResult, self).__delitem__(index)
+    
+    def __eq__(self, other):
+        self._ensureCompleteQueryResult()
+        if isinstance(other, QueryResult):
+            other._ensureCompleteQueryResult()
+        return super(QueryResult, self).__eq__(other)
+    
+    def __ne__(self, other):
+        self._ensureCompleteQueryResult()
+        if isinstance(other, QueryResult):
+            other._ensureCompleteQueryResult()
+        return super(QueryResult, self).__ne__(other)
+    
+    def __lt__(self, other):
+        self._ensureCompleteQueryResult()
+        if isinstance(other, QueryResult):
+            other._ensureCompleteQueryResult()
+        return super(QueryResult, self).__lt__(other)
+    
+    def __gt__(self, other):
+        self._ensureCompleteQueryResult()
+        if isinstance(other, QueryResult):
+            other._ensureCompleteQueryResult()
```

Comment:
If `_ensureCompleteQueryResult` function needs to be called externally, you shouldn't make it private.